rhino/src/main/java/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -1051,7 +1051,20 @@ public static String toString(Object val) {
     static String defaultObjectToString(Scriptable obj) {
         if (obj == null) return "[object Null]";
         if (Undefined.isUndefined(obj)) return "[object Undefined]";
-        return "[object " + obj.getClassName() + ']';
+
+        String tag = null;
+        if (ScriptableObject.hasProperty(obj, SymbolKey.TO_STRING_TAG)) {
+            Object tagValue = ScriptableObject.getProperty(obj, SymbolKey.TO_STRING_TAG);
```

Comment:
Maybe you can use only ScriptableObject.getProperty and then check for Scriptable.NOT_FOUND.
This will save one property lookup.

Reply:
Thanks for your feedback! Now, check Scriptable.NOT_FOUND.